### DISCUSSION TITLE
"Call to undefined function apcu_fetch()" beim Speichern der Plugin-Konfiguration (ILIAS Version 7.26)

### DISCUSSION THREAD

**Discussion by pmanheller** [studer-raimann/ViMP#27] [General]
Hallo,
wir haben das ViMP-Plugin auf unserer ILIAS Instanz (v.7.26) installiert. Beim Speichern der Plugin-Konfiguration erhalten wir folgende Fehlermeldung:
Error thrown with message "Call to undefined function apcu_fetch()"

Stacktrace:
#17 Error in /var/www/ilias/Services/GlobalCache/classes/Apc/class.ilApc.php:55
#16 ilApc:get in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/Cache/v52/class.xvmpCache.php:212
#15 xvmpCache:get in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/Model/API/class.xvmpUserRoles.php:25
#14 xvmpUserRoles:getAllAsArray in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/Model/API/class.xvmpObject.php:50
#13 xvmpObject:getAll in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/GUI/Form/class.xvmpConfFormGUI.php:282
#12 xvmpConfFormGUI:getMediaPermissionOptions in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/GUI/Form/class.xvmpConfFormGUI.php:189
#11 xvmpConfFormGUI:initForm in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/GUI/Form/class.xvmpFormGUI.php:49
#10 xvmpFormGUI:__construct in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/GUI/Form/class.xvmpConfFormGUI.php:28
#9 xvmpConfFormGUI:__construct in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/class.ilViMPConfigGUI.php:117
#8 ilViMPConfigGUI:configure in /var/www/ilias/Customizing/global/plugins/Services/Repository/RepositoryObject/ViMP/classes/class.ilViMPConfigGUI.php:64
#7 ilViMPConfigGUI:performCommand in /var/www/ilias/Services/Component/classes/class.ilPluginConfigGUI.php:81
#6 ilPluginConfigGUI:executeCommand in /var/www/ilias/Services/UICore/classes/class.ilCtrl.php:217
#5 ilCtrl:forwardCommand in /var/www/ilias/Services/Component/classes/class.ilObjComponentSettingsGUI.php:113
#4 ilObjComponentSettingsGUI:executeCommand in /var/www/ilias/Services/UICore/classes/class.ilCtrl.php:217
#3 ilCtrl:forwardCommand in /var/www/ilias/Services/Administration/classes/class.ilAdministrationGUI.php:246
#2 ilAdministrationGUI:executeCommand in /var/www/ilias/Services/UICore/classes/class.ilCtrl.php:217
#1 ilCtrl:forwardCommand in /var/www/ilias/Services/UICore/classes/class.ilCtrl.php:178
#0 ilCtrl:callBaseClass in /var/www/ilias/ilias.php:23

Nach dem ersten Speichern kann die Konfiguration nicht mehr eingesehen werden und der Fehler wird sofort geschmissen.

Könnt Ihr mir weiterhelfen?

Viele Grüße
Peter

**Comment by iljalukin** [comment]
Hallo, der APCu-Cache scheint aktiv zu sein, ist auch das entsprechende PHP-Modul installiert?

**Comment by pmanheller** [comment]
Hallo, danke für die Information. Die apcu-Erweiterung ist nun bei uns aktiviert und wir haben wieder Zugriff auf die Konfiguration.

apt-get install php-pear
pecl install apcu
/etc/php/VERSION/apache2/php.ini → extension=apcu.so eintragen
